### PR TITLE
Migrate mini cart tests

### LINKED ISSUE #42029: E2E test migration: tests/e2e-jest/specs/shopper/mini-cart.test.js
Migrate `tests/e2e-jest/specs/shopper/mini-cart.test.js` to Playwright

### PR DESCRIPTION
**Pull request by nielslange**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #42029.

Migrate the Mini Cart block e2e tests from Puppeteer to Playwright.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that all e2e tests are passing.
2. Check if the migrated tests are in a good shape or if they should be further optimised. (I haven't checked the before and after runtime, but the migrated tests run pretty fast → on average, each test executes in 2sec.)

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Migrate the Mini Cart block e2e tests from Puppeteer to Playwright.

</details>

### CODE REVIEW

**Comment by nielslange** [comment]
Closing and re-opening this PR as the CI pipeline got stuck.

Screenshot 2024-01-12 at 17 18 43

**Comment by mikejolley** [comment]
I've restarted the failing tests 🤞🏻

**Comment by nielslange** [comment]
> I've restarted the failing tests 🤞🏻

Thanks for restarting the falling test, @mikejolley. I noticed that one test, that kept failing, was related to a change I applied. I just fixed that and the tests should now pass.

File: plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.side_effects.spec.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			page.getByRole( 'link', { name: 'Mijn winkelmand bekijken' } )
+		).toBeVisible();
+
+		await expect(
+			page.getByRole( 'link', { name: 'Naar afrekenen' } )
+		).toBeVisible();
+	} );
+} );
+
+// Skipping tax tests until we sorted out a bug that hides the totals price on hover
```
Comment (mikejolley) [inline plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.side_effects.spec.ts:65]: Could you add a reference to the ticket here?

**Review by mikejolley — APPROVED** [review]
Tests look good, thanks for getting it passing.

File: plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.side_effects.spec.ts (hunk 1 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.side_effects.spec.ts:65]: Ah, good catch, @mikejolley. @Aljullu actually resolved this issue with #43550. I'll update this test accordingly.

**Comment by nielslange** [comment]
Thanks for your review, @mikejolley. I noticed one more failing e2e test, which I resolved. All tests are passing now, so I'm merging this change.

**Pull request merged**